sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -872,6 +903,18 @@ trait GpuHashJoin extends GpuExec {
     case GpuBuildRight => (right, left)
   }
 
+  // This can be overridden when a post-build project happens. When some non-ast-supported cases
+  // exist in join condition, it will be extracted from join condition and wrapped as pre-project
+  // nodes.
+  protected lazy val joinLeftOutput = left.output;
+  protected lazy val joinRightOutput = right.output;
```

Comment:
Rather than have this somewhat redundant API, should we just have GpuHashJoin require an AST join strategy and leverage those APIs?  Seems like there's extra code and complexity here because we're avoiding that.

Reply:
Removed with a refactor